packages/v4/src/content/design-guidelines/components/tree-view/tree-view.md

```diff
@@ -3,5 +3,92 @@ id: Tree view
+* The data is structured into levels.
+* You need to display hierarchies that have more than 2 levels.
+
+Don’t use a tree view when: 
+* The items are not structured hierarchically. Use the [data list](/components/data-list) instead. 
+* You select 1  item from a small set. Instead, use [select lists](/), [radio buttons](/components/radio), [checkboxes](/components/checkbox), or a [dropdown](/components/dropdown).
+* You need to see an overview of a large amount of data.
+* You need to compare two large sets of data. Use [charts](/charts/about) for a better view. 
+* The hierarchy is too deep and includes hundreds of items.
+
+## Tree view capabilities
+
+You can add these functionalities to a tree view:
+* **[Search bar](#adding-a-search-bar-to-a-tree-view)** for searching items in the tree view
+* **[Checkboxes](#adding-checkboxes-to-a-tree-view)** for selecting items in the tree view
+* **[Badges](#adding-badges-to-a-tree-view)** for showing the number of child nodes in the tree view
+* **[Icons](#adding-icons-to-a-tree-view)** for visually representing the node types in the tree view
+* **[Action items](#adding-action-items-to-a-tree-view)** for adding an action to each item in the tree view
+
+### Adding a search bar to a tree view
+
+Add a search bar to a tree view if you're including a large amount of data and need to provide users with a quick way to find a single item. If your dataset is small and easy to navigate, then you don't need to add a search bar.
+
+<img src="./img/with-search-bar.png" width="940" />
+
+### Adding checkboxes to a tree view
+
+Add checkboxes to a tree view if you’re selecting multiple items or applying multiple filters. Selecting should be a primary function of your use-case. If your dataset is small and you select just 1 item, use [select lists](/components/select), [radio buttons](/components/radio), [checkboxes](/components/checkbox), or a [dropdown](/components/dropdown) instead. 
+
+<img src="./img/with-checkboxes.png" width="932" />
+
+### Adding badges to a tree view
+
+Add badges to a tree view if you want to show the number of items inside. Don’t use it for displaying different information. The last item in a tree will not have a badge because it has 0 children. 
+
+<img src="./img/with-badges.png" width="332" />
+
+### Adding icons to a tree view
+
+Add icons to a tree view if you’re visually representing the type of items in a tree. For example, the folder system. Apply icons on all items in a tree to keep consistency. We don’t recommend to use icons and checkboxes at the same time due to large indentation and loss of space. 
+
+<img src="./img/with-icons.png" width="332" />
+
+### Adding action items to a tree view
+
+Add action items to a tree view if you’re triggering some type of action above the items. These actions appear on hover to avoid overcrowding the tree view. We recommend using 1 type of action across the whole tree and the maximum of actions for 1 item is 1.
```

Comment:
Would reword "We recommend using 1 type of action across the whole tree and **to limit** the maximum of actions for 1 item **to** 1."